cmd/syslog/syslog.go

```diff
@@ -20,14 +23,41 @@ func main() {
 	server.SetFormat(syslog.RFC3164)
 	server.SetHandler(handler)
 
+	// check if syslog unix socket is already there then delete it.
+	if _, err := os.Stat(listenAddress); err == nil {
+		if err := os.Remove(listenAddress); err != nil {
+			log.Fatalf("error removing syslog socket %s: %s", listenAddress, err)
+		}
+	}
+
 	if err := server.ListenUnixgram(listenAddress); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to listen to syslog unix socket %s: %s", listenAddress, err)
 	}
 
+	defer func() {
+		if err := os.Remove(listenAddress); err != nil {
+			log.Fatalf("error removing syslog socket %s: %s", listenAddress, err)
+		}
+	}()
+
 	if err := server.Boot(); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to boot syslog server: %s", err)
 	}
 
+	// Unix sockets must be unlink()ed before being reused again.
+	// Handle common process-killing signals so we can gracefully shut down:
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
+	go func(c chan os.Signal) {
+		// Wait for a SIGINT or SIGKILL:
+		sig := <-c
+		log.Printf("Caught signal %s: shutting down.\n", sig)
+		// Stop listening (and unlink the socket if unix type):
+		if err := server.Kill(); err != nil {
+			log.Printf("Failed to close server connections: %s.\n", err)
```

Comment:
I think this should be fatal. You really want to exit. If the call to kill fails you'll still be stuck in `server.Wait()`.